src/containers/ViewerPanel/test/selectors.test.ts

```diff
@@ -70,7 +74,95 @@ describe("ViewerPanel selectors", () => {
                 roundedTime: 6,
                 roundedLastFrameTime: 30,
                 roundedTimeStep: 0.2,
+                maxNumChars: 4,
             });
         });
+
+        it("shows correct maxNumChars when firstFrameTime is very long", () => {
+            const mockState: State = {
+                ...initialState,
+                selection: {
+                    ...initialState.selection,
+                    time: 3.0001,
+                },
+                metadata: {
+                    ...initialState.metadata,
+                    timeUnits: {
+                        magnitude: 1,
+                        name: "ns",
+                    },
+                    timeStep: 0.0800000000023,
+                    lastFrameTime: 15.0001,
+                    firstFrameTime: 0.000004000000001,
+                },
+            };
+
+            const displayTimes = getDisplayTimes(mockState);
+
+            expect(displayTimes).to.deep.equal({
+                roundedTime: 3,
+                roundedLastFrameTime: 15,
+                roundedTimeStep: 0.08,
+                maxNumChars: 8,
+            });
+        });
+    });
+
+    describe("getMaxNumChars", () => {
+        it("determines correct maxNumChars when time values are simple integers", () => {
```

Comment:
nit but I think it would be helpful to define what is "correct maxNum" here. Like, "if time values are simple integers, returns a max based on the largest value time can possibly be"

Reply:
Good call, I made them better